controllers/promos.ts

```diff
@@ -24,10 +27,29 @@ export const getActivePromos = asyncHandler(async (req, res, next) => {
   const activePromosWithContentMetadata = await Promise.all(
     activePromos.map(async (promo) => {
       const contentMetadata = await getContentInfoFromId(promo.contentId);
+      const totalEarned = await getTotalPromoEarnedByUser(accountId, promo.id);
+      const totalEarnedToday = await getTotalPromoEarnedByUserToday(
+        accountId,
+        promo.id
+      );
+
       if (!contentMetadata) {
         return;
       }
```

Comment:
Maybe switch ordering to return sooner?

```suggestion
      if (!contentMetadata) {
        return;
      }
      
      const totalEarned = await getTotalPromoEarnedByUser(accountId, promo.id);
      const totalEarnedToday = await getTotalPromoEarnedByUserToday(
        accountId,
        promo.id
      );
```